tests/test_typing/bad/err3.catala_en

```diff
@@ -17,17 +17,17 @@ $ catala Typecheck
 --> decimal
 
 Error coming from typechecking the following expression:
-  --> tests/test_typing/bad/common.catala_en
+  --> tests/test_typing/bad/err3.catala_en
    |
-15 |   output a content decimal
-   |          ^
+10 |   definition a equals number of (z ++ z) / 2
+   |                                          ^
    +
 
 Type integer coming from expression:
   --> tests/test_typing/bad/err3.catala_en
    |
 10 |   definition a equals number of (z ++ z) / 2
-   |                       ^^^^^^
+   |                                          ^
```

Comment:
So generally the type indicating the operator is not the most obvious hint as to what's going wrong in your program. But I guess we can live with it, until we get our OCaml version of https://docs.rs/ariadne/latest/ariadne/index.html and are able to craft magnificent error messages with notes, hints and stuff.

Reply:
I agree. I must admit I have been turning some knobs to tweak the error messages to better positions, but we're not completely there yet. It's also a notoriously difficult problem where you have to account for how people think :)

Here the type mismatch comes from the return type of the operator: the bug is that we should actually mark the whole application and not just the operator (the `/` expression is *not* actually of type int)! I didn't yet understand exactly why, but I'd argue that marking `/` already makes more sense than marking `number`.

Still, I am glad that the expression marked as the source of the error is now the actual expression and not a part of the declaration in the metadata section!

Reply:
After more detailed investigation, this comes from the fact that unification is done on the arrow type: we can build that type from the outer information and the arguments, then do a single unification. It makes sense then that this points to the operator. This would likely be non-trivial to change…